inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/src/matchers/single_op.cpp

```diff
@@ -0,0 +1,155 @@
+bool compare_constants_data(const std::shared_ptr<ngraph::op::Constant> &op,
+                            const std::shared_ptr<ngraph::op::Constant> &ref) {
+    size_t elements_count = ngraph::shape_size(op->get_shape());
+    if (elements_count != ngraph::shape_size(ref->get_shape())) {
+        return false;
+    }
+    const auto &op_data = op->cast_vector<dType>();
+    const auto &ref_data = ref->cast_vector<dType>();
+    for (size_t i = 0; i < elements_count; ++i) {
+        // std:abs doesn't implemented for unsigned types, compare explicitly to keep code universal for all dTypes
+        dType diff = op_data[i] - ref_data[i] > 0 ? op_data[i] - ref_data[i] : ref_data[i] - op_data[i];
+        if (diff > std::numeric_limits<dType>::epsilon()) {
+            return false;
+        }
+    }
+    return true;
+}
+
+// TODO: Move to some utils?
+bool compare_constants_data(const std::shared_ptr<ngraph::op::Constant> &op,
+                            const std::shared_ptr<ngraph::op::Constant> &ref) {
+    switch (op->get_element_type()) {
+        case ngraph::element::Type_t::boolean:
+            return compare_constants_data<bool>(op, ref);
+        case ngraph::element::Type_t::bf16:
+            return compare_constants_data<ngraph::bfloat16>(op, ref);
+        case ngraph::element::Type_t::f16:
+            return compare_constants_data<ngraph::float16>(op, ref);
+        case ngraph::element::Type_t::f32:
+            return compare_constants_data<float>(op, ref);
+        case ngraph::element::Type_t::f64:
+            return compare_constants_data<double>(op, ref);
+        case ngraph::element::Type_t::i8:
+            return compare_constants_data<int8_t>(op, ref);
+        case ngraph::element::Type_t::i16:
+            return compare_constants_data<int16_t>(op, ref);
+        case ngraph::element::Type_t::i32:
+            return compare_constants_data<int32_t>(op, ref);
+        case ngraph::element::Type_t::i64:
+            return compare_constants_data<int64_t>(op, ref);
+        // TODO cast_vector doesn't support u1 now
+//        case ngraph::element::Type_t::u1:
+//            return compare_constants_data<char>(op, ref);
+        case ngraph::element::Type_t::u8:
+            return compare_constants_data<uint8_t>(op, ref);
+        case ngraph::element::Type_t::u16:
+            return compare_constants_data<uint16_t>(op, ref);
+        case ngraph::element::Type_t::u32:
+            return compare_constants_data<uint32_t>(op, ref);
+        case ngraph::element::Type_t::u64:
+            return compare_constants_data<uint64_t>(op, ref);
+        default:
+            std::cerr << "Can't compare constants" << op << " with " << ref << "\n" << "Unsupported data type";
+            return false;
+    }
+}
+
+const char *SingleOpMatcher::name = "generic_single_op";
+
+bool SingleOpMatcher::match(const std::shared_ptr<ngraph::Node> &node, const std::shared_ptr<ngraph::Node> &ref) {
+    // Match OP type and version
```

Comment:
When I see comments like this I always ask myself why is not a function.
This might looks like:
```
return sameOpType(n1,n2) && 
       matchInputs(n1, n2) && 
       matchOutputs(n1, n2) && 
       sameAttributes(n1, n2) &&
       matchPorts(n1, n2);
```
Or even better:
```
return NodeComparator(n1, n2).modes_match();
```

Reply:
Good point, I prefer to stick to the first approach.
Second one doesn't really fit to this comparators usage scenario - In theory there can be different realisation of matchers with different uniqueness criteria even for a single operation, that's why I think that having one NodeComparator  including all matching rules will not really work as expected from Matchers architecture perspective.

But splitting match function to the set of private members performing certain type of comparison is a definitely good idea

Reply:
Created set of private members of SingleOPMatcher for performing each type of matching